### DISCUSSION TITLE
Doc for --disableGitFeatures doesn't explain what the "git features" are

### DISCUSSION THREAD

**Discussion by jehumb-havok** [dotnet/docfx#8275] [General]
**Operation System**: `Windows`

**DocFX Version Used**: 2.51.0 (NuGet)

**Template used**: `default`

**Steps to Reproduce**:

1. Run docfx on a git repository

**Expected Behavior**:

Documentation explains what the "git features" are, so user can know whether having them enabled is of interest or not.

**Actual Behavior**:

Documentation [simply says `--disableGitFeatures` is to "Disable fetching Git related information for articles"](https://dotnet.github.io/docfx/tutorial/docfx.exe_user_manual.html#32-properties-for-build), and user doesn't know what features they are losing by using that flag.

Incidentally, running with:
```shell
docfx.exe docfx.json --intermediateFolder ..\build\docs\obj -o ..\build\docs  --cleanupCacheHistory --debugOutput --debug  --disableGitFeatures
```
I still can hit some git error:
```shell
Skipping GetFileDetail. Exception found: Microsoft.DocAsCode.Common.Git.GitException, Message:
[...] can't find remote branch in this repo and fallback to use local branch [master]: fatal: no upstream configured for branch 'master'
```

**Comment by QFSW** [comment]
My understanding was that git features allows you to jump to source code from API reference, but that may be incorrect/incomplete (so you're definitely right the documentation needs improving)

**Comment by KalleOlaviNiemitalo** [comment]
In DocFX v2.50, the `disableGitFeatures` flag is checked (as opposed to being propagated) only by [GitUtility.TryGetFileDetail](https://github.com/dotnet/docfx/blob/ac8024bc50bd47d748ff99791a70e6f1ef6152d8/src/Microsoft.DocAsCode.Common/Git/GitUtility.cs#L53-L71), which collects the remote repository URL and branch name. The template then [generates the URL](https://github.com/dotnet/docfx/blob/ac8024bc50bd47d748ff99791a70e6f1ef6152d8/src/docfx.website.themes/common/common.js) for the "Improve this Doc" link from those.

In DocFX v3, there is instead code to get the GitHub login names of people who contributed to the documentation. Which can't work if the repository is hosted elsewhere.
